### ISSUE TITLE
Vehicles with hard speed cap [Improvement]: 

### ISSUE THREAD

**Issue by clickisbored** [Yimura/YimMenu#3277]
### Which feature

As of right now the vehicle tab can make most cars go to the 300+ max speed cap of the game, but some cars like my Dominator ASP and especially electric cars are limited to a speed of 120-130. I have no idea if its even possible to remove the hard caps on these vehicles but if it may be possible I would love to see it implemented into the menu or even just a personal request for myself if this is noticed. Thank you

### Reason

Limited top speed of specific cars make some of my favorite cars far too boring.

### Additional context

![Screenshot 2024-06-21 221852](https://github.com/YimMenu/YimMenu/assets/173325696/5c8eb94d-b32c-41b1-9e26-f4f6eb7770cb)

**Comment by clickisbored** [comment]
Don't know anything about lua or c++ or coding in general... dumb it down for me.

**Comment by Rxann** [comment]
This would likely be implemented in lua and not in the c++ part of the menu.

**Comment by clickisbored** [comment]
> This would likely be implemented in lua and not in the c++ part of the menu.

I figured but when I go looking for lua's its usually not what im looking for or someone who just gave up on it, which is why i'm not sure it's fully possible. Any help would be great!

**Comment by Rxann** [comment]
> > This would likely be implemented in lua and not in the c++ part of the menu.
> 
> I figured but when I go looking for lua's its usually not what im looking for or someone who just gave up on it, which is why i'm not sure it's fully possible. Any help would be great!

There is some good documentation for the YimMenu lua API and lua is a pretty simple language to learn.

**Comment by xesdoog** [comment]
[TokyoDrift](https://github.com/xesdoog/TokyoDrift) has NOS which increases the top speed of your vehicle when you hold [**Left Shift**]

**Comment by binded-zz** [comment]
have you tried idk if it works but should 

https://github.com/Harmless05/harmless-lua

he has a set max speed for vehicles

**Comment by clickisbored** [comment]
> have you tried idk if it works but should
> 
> https://github.com/Harmless05/harmless-lua
> 
> he has a set max speed for vehicles

I have tried the Lua and currently have it installed, I think its for cars that can already go faster than 125 as it doesn't work for unlocking the speed limiter rockstar built into the cars.

**Comment by clickisbored** [comment]
> [TokyoDrift](https://github.com/xesdoog/TokyoDrift) has NOS which increases the top speed of your vehicle when you hold [**Left Shift**]

This works pretty well actually, although my speed is lost rather fast as soon as I let go its make the car much faster. Thank you!

**Comment by xesdoog** [comment]
> my speed is lost rather fast as soon as I let go

Yes I wrote it that way intentionally to make it simulate NOS like in some racing games (Need For Speed for example) but with a few edits it could be used differently. I'll probably take a look at [Harmless](https://github.com/Harmless05/harmless-lua) top speed option and make it always work on all cars (TokyoDrift was built on one of Harmless's scripts)

**Comment by clickisbored** [comment]
I'll probably take a look at [Harmless](https://github.com/Harmless05/harmless-lua) top speed option and make it always work on all cars (TokyoDrift was built on one of Harmless's scripts)

Maybe you could add settings for what speed and strength the nos works at automatically. I believe that would almost entirely solve my issue.

**Comment by clickisbored** [comment]
> > my speed is lost rather fast as soon as I let go
> 
> Yes I wrote it that way intentionally to make it simulate NOS like in some racing games (Need For Speed for example) but with a few edits it could be used differently. I'll probably take a look at [Harmless](https://github.com/Harmless05/harmless-lua) top speed option and make it always work on all cars (TokyoDrift was built on one of Harmless's scripts)

Maybe you could add settings for what speed and strength the nos works at automatically. I believe that would almost entirely solve my issue.

also I'm sure the car slowing down when I let go is because the game is forcing the car back down to the 125 speed cap.

**Comment by xesdoog** [comment]
> also I'm sure the car slowing down when I let go is because the game is forcing the car back down to the 125 speed cap.

It slows down because the script is telling the game to reset the vehicle to stock power when you let go of **Left Shift**

![image](https://github.com/YimMenu/YimMenu/assets/66764345/11d02ceb-5020-47ef-92ea-b7bb81d50fa0)
